sparse_ho/ho.py

```diff
@@ -13,34 +13,41 @@ def grad_search(
 
     Parameters
     ----------
-    algo: TODO
-    criterion: TODO
-    model: TODO
+    algo: BaseAlgo instance
+        algorithm used to compute hypergradient.
+    criterion: BaseCriterion
+        criterion to optimize during hyperparameter optimization
+        (outer optimization problem).
+    model: BaseModel
+        model on which hyperparameter has to be selected
+        (inner optimization problem).
+    X: {ndarray, sparse matrix} of (n_samples, n_features)
+        Data.
+    y: {ndarray, sparse matrix} of shape (n_samples,)
+        Target.
     log_alpha0: float
-        log of the regularization coefficient alpha
-    tol : float
-        tolerance for the inner optimization solver
-    monitor: Monitor object
-        used to store the value of the cross-validation function
+        log of the regularization coefficient alpha.
+    monitor: instance of Monitor
+        used to store the value of the cross-validation function.
     n_outer: int
-        number of maximum iteration in the outer loop (for the line search)
+        Defaults to 100.
+        number of maximum iteration in the outer loop (for the line search).
```

Comment:
iterations
number of maximum line search iterations ? Or number of updates of alpha?